### PR TITLE
[IMP] account, web, base: OWL migration 1

### PR DESCRIPTION
**Pull request by h4818**
Migrate account javascript to owl

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/core/domain.js [hunk 1]
```diff
@@ -206,6 +206,9 @@ function matchCondition(record, condition) {
             return fieldValue === value;
         case "!=":
         case "<>":
+            if (Array.isArray(fieldValue) && Array.isArray(value)) {
+                return !shallowEqual(fieldValue, value);
+            }
```
Comment (aab-odoo) [inline addons/web/static/src/core/domain.js:211]: Can you provide the input values for which this function wasn't working as you expect? I guess one of the value if `false`? I'm not sure this function should handle such a case, as false isn't a valid x2many values (the value of an unset x2many is the empty array).

Comment (h4818) [reply addons/web/static/src/core/domain.js:211]: Hi @aab-odoo 
This is the domain thats not working:
https://github.com/odoo/odoo/blob/master/addons/payment/views/account_invoice_views.xml#L13
`('authorized_transaction_ids', '!=', [])`

Comment (aab-odoo) [reply addons/web/static/src/core/domain.js:211]: Ok but what was the value of `authorized_transaction_ids` in your case? Or could you give me the steps to reproduce your issue on a runbot?

Comment (h4818) [reply addons/web/static/src/core/domain.js]: https://github.com/odoo/odoo/pull/96614

**Comment by h4818** [comment]
https://github.com/odoo/odoo/pull/96672

**Pull request closed without merging**